packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -146,32 +263,171 @@ class TimeInput extends React.Component {
+    this.setState({
+      meridiem: event.target.value,
+    });
+
+    this.handleValueChange(event, inputType.HOUR, this.state.hour.toString(), event.target.value);
   }
 
-  handleInputKeyDown(event, type) {
-    let stateValue = type === inputType.HOUR ? this.state.hour : this.state.minute;
-    const maxValue = type === inputType.HOUR ? 23 : 59;
+  meridiemFocus() {
+    this.setState({
+      isFocused: true,
+      meridiemFocused: true,
+    });
+
+    this.meridiemInput.focus();
+  }
+
+  meridiemBlur() {
+    this.setState({
+      isFocused: false,
+      meridiemFocused: false,
+    });
+  }
+
+  handleHourInputKeyDown(event) {
+    let stateValue = this.state.hour;
+    const maxValue = this.props.variant === FORMAT_12_HOUR ? 12 : 23;
+    const minValue = this.props.variant === FORMAT_12_HOUR ? 1 : 0;
+
+    let meridiem = this.state.meridiem;
+    let updateStateValue = false;
+
+    if (event.keyCode === keyCodes.ARROWUP) {
+      // Increment the value by 1 when arrow up is pressed.
+      if (stateValue) {
+        let numericMinute = Number(stateValue);
+
+        if (numericMinute < maxValue) {
+          numericMinute += 1;
+
+          if (numericMinute < 10) {
+            stateValue = '0'.concat(numericMinute.toString());
+          } else {
+            stateValue = numericMinute.toString();
+          }
+          updateStateValue = true;
+
+          // Hitting 12 when incrementing hour changes the meridiem
+          if (this.props.variant === FORMAT_12_HOUR && numericMinute === 12) {
+            meridiem = meridiem === this.postMeridiem ? this.anteMeridiem : this.postMeridiem;
+          }
+        } else if (this.props.variant === FORMAT_12_HOUR) {
+          stateValue = '01';
+          updateStateValue = true;
+        }
+      } else {
+        stateValue = this.props.variant === FORMAT_12_HOUR ? '12' : '00';
+        updateStateValue = true;
+      }
+    } else if (event.keyCode === keyCodes.ARROWDOWN) {
+      // Decrement the value by 1 when arrow down is pressed.
+      if (stateValue) {
+        let numericMinute = Number(stateValue);
+
+        if (numericMinute > minValue) {
+          numericMinute -= 1;
+
+          if (numericMinute < 10) {
+            stateValue = '0'.concat(numericMinute.toString());
+          } else {
+            stateValue = numericMinute.toString();
+          }
+          updateStateValue = true;
+
+          // Hitting 11 when decrementing hour changes the meridiem
+          if (this.props.variant === FORMAT_12_HOUR && numericMinute === 11) {
+            meridiem = meridiem === this.postMeridiem ? this.anteMeridiem : this.postMeridiem;
+          }
+        } else if (this.props.variant === FORMAT_12_HOUR) {
+          stateValue = '12';
+          updateStateValue = true;
+        }
+      } else {
+        stateValue = this.props.variant === FORMAT_12_HOUR ? '12' : '00';
+        updateStateValue = true;
+      }
+    } else if (event.keyCode === keyCodes.ARROWRIGHT) {
```

Comment:
What are your thoughts on breaking this off to its own if statement:

`if (event.keyCode === keyCodes.ARROWRIGHT) {` instead of using `else if`

Reply:
Done here. https://github.com/cerner/terra-core/pull/1027/commits/45b44e889215a8dec2e1aed334cdb59ae3670902